homeassistant/components/hvac/__init__.py

```diff
@@ -481,13 +484,13 @@ def max_humidity(self):
 
     def _convert_for_display(self, temp):
         """Convert temperature into preferred units for display purposes."""
-        if temp is None:
-            return None
+        if temp is None or not isinstance(temp, Number):
```

Comment:
I don't think it can ever be `not isinstance(temp, Number)`. And if it is, we should probably have convert_temperature blow up.